src/target/riscv/batch.h

```diff
@@ -25,40 +25,64 @@ enum riscv_scan_delay_class {
 	/* Delay for System Bus read operation: */
 	RISCV_DELAY_SYSBUS_READ,
 	/* Delay for System Bus write operation: */
-	RISCV_DELAY_SYSBUS_WRITE,
+	RISCV_DELAY_SYSBUS_WRITE
 };
 
+static inline const char *
+riscv_scan_delay_class_name(enum riscv_scan_delay_class delay_class)
+{
+	switch (delay_class) {
+	case RISCV_DELAY_BASE:
+		return "DM access";
+	case RISCV_DELAY_ABSTRACT_COMMAND:
+		return "Abstract Command";
+	case RISCV_DELAY_SYSBUS_READ:
+		return "System Bus read";
+	case RISCV_DELAY_SYSBUS_WRITE:
+		return "System Bus write";
+	}
+	assert(0);
+	return NULL;
+}
+
+/* The purpose of these delays is to be passed to "jtag_add_runtest()",
+ * which accepts an "int".
+ * Therefore, they should be no greater then "INT_MAX" after adding the base
+ * delay.
+ */
```

Comment:
I have tried to reword the comment for to make it even clearer. Please let me know if you like this wording.

```suggestion
/* The scan delay values are passed to "jtag_add_runtest()", which accepts an "int".
 * Therefore, the passed value should be no greater than "INT_MAX".
 *
 * Since the resulting delay value can be a sum of two individual delays, we limit the 
 * individual delays to INT_MAX / 2 to prevent overflow of the final sum.
 */
```

Reply:
I believe it's better, thanks! Addressed.